flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/FlinkJoinToMultiJoinRule.java

```diff
@@ -178,6 +193,60 @@ public void onMatch(RelOptRuleCall call) {
         call.transformTo(multiJoin);
     }
 
+    private void buildInputNullGenFieldList(
+            RelNode left, RelNode right, JoinRelType joinType, List<Boolean> isNullGenFieldList) {
+        if (joinType == JoinRelType.INNER) {
+            buildNullGenFieldList(left, isNullGenFieldList);
+            buildNullGenFieldList(right, isNullGenFieldList);
+        } else if (joinType == JoinRelType.LEFT) {
+            // left judge.
+            buildNullGenFieldList(left, isNullGenFieldList);
+
+            for (int i = 0; i < right.getRowType().getFieldCount(); i++) {
+                isNullGenFieldList.add(true);
+            }
+        } else if (joinType == JoinRelType.RIGHT) {
+            for (int i = 0; i < left.getRowType().getFieldCount(); i++) {
+                isNullGenFieldList.add(true);
+            }
+
+            // right judge.
+            buildNullGenFieldList(right, isNullGenFieldList);
+        } else if (joinType == JoinRelType.FULL) {
+            for (int i = 0; i < left.getRowType().getFieldCount(); i++) {
+                isNullGenFieldList.add(true);
+            }
+            for (int i = 0; i < right.getRowType().getFieldCount(); i++) {
+                isNullGenFieldList.add(true);
+            }
+        }
```

Comment:
nit: it's better we could add some comments to explain what's the behavior is the `else` branch

Reply:
> nit: it's better we could add some comments to explain what's the behavior is the `else` branch

Done!

Reply:
we should throw exception for `else` branch

Reply:
Done!